rust-runtime/smithy-json/src/serialize.rs

```diff
@@ -0,0 +1,451 @@
+        self.json.push_str(match value {
+            true => "true",
+            _ => "false",
+        });
+        self
+    }
+
+    /// Writes a string `value` with the given `key`.
+    pub fn string(&mut self, key: &str, value: &str) -> &mut Self {
+        self.key(key);
+        append_string(&mut self.json, value);
+        self
+    }
+
+    /// Writes a string `value` with the given `key` without escaping it.
+    pub fn trusted_string(&mut self, key: &str, value: &str) -> &mut Self {
+        self.key(key);
+        append_string(&mut self.json, value);
+        self
+    }
+
+    /// Writes a number `value` with the given `key`.
+    pub fn number(&mut self, key: &str, value: Number) -> &mut Self {
+        self.key(key);
+        append_number(&mut self.json, value);
+        self
+    }
+
+    /// Writes an Instant `value` with the given `key` and `format`.
+    pub fn instant(&mut self, key: &str, instant: &Instant, format: Format) -> &mut Self {
+        self.key(key);
+        append_instant(&mut self.json, instant, format);
+        self
+    }
+
+    /// Starts an array with the given `key`.
+    pub fn start_array(&mut self, key: &str) -> JsonArrayWriter {
+        self.key(key);
+        JsonArrayWriter::new(&mut self.json)
+    }
+
+    /// Starts an object with the given `key`.
+    pub fn start_object(&mut self, key: &str) -> JsonObjectWriter {
+        self.key(key);
+        JsonObjectWriter::new(&mut self.json)
+    }
+
+    /// Finishes the object.
+    pub fn finish(self) {
+        self.json.push('}');
+    }
+
+    fn key(&mut self, key: &str) {
+        if self.started {
+            self.json.push(',');
+        }
+        self.started = true;
+
+        self.json.push('"');
+        self.json.push_str(&escape_string(key));
+        self.json.push_str("\":");
+    }
+}
+
+pub struct JsonArrayWriter<'a> {
+    json: &'a mut String,
+    started: bool,
+}
+
+impl<'a> JsonArrayWriter<'a> {
+    pub fn new(output: &'a mut String) -> Self {
+        output.push('[');
+        Self {
+            json: output,
+            started: false,
+        }
+    }
+
+    /// Writes a null value to the array.
+    pub fn null(&mut self) -> &mut Self {
+        self.comma_delimit();
+        self.json.push_str("null");
+        self
+    }
+
+    /// Writes the boolean `value` to the array.
+    pub fn boolean(&mut self, value: bool) -> &mut Self {
+        self.comma_delimit();
+        self.json.push_str(match value {
+            true => "true",
+            _ => "false",
+        });
+        self
+    }
+
+    /// Writes a string to the array.
+    pub fn string(&mut self, value: &str) -> &mut Self {
+        self.comma_delimit();
+        append_string(&mut self.json, value);
+        self
+    }
+
+    /// Writes a string `value` to the array without escaping it.
+    pub fn trusted_string(&mut self, value: &str) -> &mut Self {
+        self.comma_delimit();
+        append_string(&mut self.json, value);
```

Comment:
should be `append_trusted_string`
```suggestion
        append_trusted_string(&mut self.json, value);
```

Reply:
I would probably renamed this `string_unchecked` — it's not that we trust/don't trust the string as much that we aren't going to check that it doesn't need to be escaped